tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -856,11 +889,30 @@ def test_split_observation_pairs() -> None:
             (-float("inf"), [-5.0]),  # COMPLETE
         ],
         2,
+        None,
     )
     assert list(indices_below) == [0, 3]
     assert list(indices_above) == [1, 2]
 
 
+def test_split_observation_pairs_with_constraints() -> None:
+    indices_below, indices_above = _tpe.sampler._split_observation_pairs(
+        [
+            (-7, [-2]),  # PRUNED (with intermediate values)
+            (float("inf"), [0.0]),  # PRUNED (without intermediate values)
+            (
+                -3,
+                [float("inf")],
+            ),  # PRUNED (with a NaN intermediate value; it's treated as infinity)
+            (-float("inf"), [-5.0]),  # COMPLETE
```

Comment:
The comments look strange. Could you update them as they are well-described?

Reply:
How are the comments strange? It's actually copy-pasted from the above function `test_split_observation_pairs()`.